sunbeam-python/sunbeam/provider/local/deployment.py

```diff
@@ -61,7 +62,7 @@
 
 
 class LocalDeployment(Deployment):
-    name: str = "local"
+    name: str = petname.Generate()
```

Comment:
Won't this change at every instanciation of LocalDeployment? and how does this work on a multi node local deployment?

Reply:
The deployment is loaded using [1] which picks the .local/share/openstack/deployment.yaml if it exists. So instantiation should be fine unless the deployment.yaml file is removed.

Currently the deployment name is used only to determine cloud name, credential name which comes during bootstrap phase - creation of manual cloud, k8s cloud, credential, creation of machine model. 
Multi node should work since cloud/credentials are not referred.
However running some of the sunbeam commands on non primary node will fail like enabling observability feature. Currently i believe certain commands running on non primary node might fail as the juju user does not have access to certain juju objects like offers. 

[1] https://github.com/canonical/snap-openstack/blob/main/sunbeam-python/sunbeam/main.py#L117

Reply:
As discussed, deployment name is uploaded to clusterdb and read on non-primary nodes in join command.